tests/callbacks/test_model_checkpoint.py

```diff
@@ -204,6 +204,30 @@ def test_model_checkpoint_save_last_checkpoint_contents(tmpdir):
         assert w0.eq(w1).all()
 
 
+def test_model_checkpoint_none_monitor(tmpdir):
+    model = EvalModelTemplate()
+    epochs = 2
+    checkpoint_callback = ModelCheckpoint(filepath=tmpdir, monitor=None, save_top_k=-1)
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        early_stop_callback=False,
+        checkpoint_callback=checkpoint_callback,
+        max_epochs=epochs,
+    )
+    trainer.fit(model)
+
+    # these should not be set if monitor is None
+    assert checkpoint_callback.best_model_path == ''
```

Comment:
just thinking why not saving the best model, as a user, I am always interested in the best one without spending much time looking in charts which was the best...